db/sql/src/protocol.rs

```diff
@@ -236,7 +242,8 @@ impl HoprDbProtocolOperations for HoprDb {
                 plain_text,
                 ..
             } => {
-                let ack = Acknowledgement::new(ack_key, pkt_keypair);
+                let offchain_keypair = pkt_keypair.clone();
+                let ack = spawn_fifo_blocking(move || Acknowledgement::new(ack_key, &offchain_keypair)).await;
```

Comment:
A general question : How does it know which thread pool to be used?

Reply:
Our parallelization has one. And it schedules all jobs on top of that.